metadata-ingestion/src/datahub/ingestion/source/kafka.py

```diff
@@ -65,9 +66,18 @@ def __init__(self, config: KafkaSourceConfig, ctx: PipelineContext):
                 **self.source_config.connection.consumer_config,
             }
         )
-        self.schema_registry_client = SchemaRegistryClient(
-            {"url": self.source_config.connection.schema_registry_url}
-        )
+        logger.info(f"Entering modified kafka.py {self.source_config.connection.schema_registry_class}")
+        """
+        schema_registry_class = "datahub.ingestion.source.ConfluentSchemaRegistry"
+        if(self.source_config.connection.schema_registry_class != "confluent"):
+            self.schema_registry_client =  SchemaRegistryClient(
+                {"url": self.source_config.connection.schema_registry_url}
+            )
+        else:
+            self.source_config.connection.schema_registry_class
+        """
+        self.schema_registry_client = pydantic_resolve_key(self.source_config.connection.schema_registry_class)
```

Comment:
Is this supposed to be the `SchemaRegistryClient` class or a Factory that hands you back a `SchemaRegistryClient`?